nuxeo-ai-model/src/main/java/org/nuxeo/ai/model/serving/SuggestionOp.java

```diff
@@ -61,7 +83,82 @@ public Blob run(DocumentModel doc) throws IOException {
         if (doc == null || (suggestions = modelServingService.predict(doc)) == null || suggestions.isEmpty()) {
             return Blobs.createJSONBlob(EMPTY_JSON_LIST);
         }
-        return Blobs.createJSONBlob(MAPPER.writeValueAsString(suggestions));
+
+        String toReturn = toJsonString(jg -> {
+            // JsonGenerator won't let you start with an array, so I am wrapping it in a field and removing the
+            // field at the end of the method.
+            jg.writeFieldName("return");
+            jg.writeStartArray();
+            suggestions.forEach(metadata -> {
+                try {
+                    DocumentModel input = coreSession.getDocument(new IdRef(metadata.context.documentRef));
```

Comment:
just for performance reasons I would consider `CoreSession#getDocuments` here, to basically pre-fetch. 
Although we know what there will be a small amount of suggestion, so not critical. Just a note